README.md

```diff
@@ -5,14 +5,25 @@
 
 ## Routes
 
-### `GET /retrieval`
+### `PUT /retrieval`
+
+Start a new retrieval.
 
 Response:
 
-```json
-{}
+```typescript
+{
+  id: String,
+  cid: String,
+  providerAddress: String,
+  protocol: 'graphsync'|'bitswap'
+}
 ```
 
+### `POST /retrieval/:id`
+
+_TODO_
+
 ## Development
 
 1. Set up [PostgreSQL](https://www.postgresql.org/) with default settings:
```

Comment:
>  - Port: 5432
>  - User: _your user name_
>  - Password: _blank_
>  - Database: _same as user name_

This comment is probably out of scope of this PR, but I am not very comfortable with this setup.

What is "my username" - is it the same as printed by `echo $USER`? I want to use a different username & database for each project. I'd also like to us a non-empty password.

What privileges must the DB user have? Ideally, I want to use a project-specific user that has access to the project-specific database only. That means that as part of the dev setup, I need to create the database & the user on my PG server. 

It would be great to have a script I can run to set everything up for me.

Alternatively, we can run PG via Docker. That saves us from a ton of complexity. For example, to nuke the DB and start from scratch, we can simply delete the old Docker container and create a new one.

In the past, I was using npm package scripts like these:

```json
{
  "scripts": {
    "db:start": "docker run --name spark-db -p 5432:5432 -e POSTGRES_PASSWORD=spark-space-traveller -d ghcr.io/baosystems/postgis:13-3.1 -c shared_preload_libraries='pg_stat_statements'",
    "db:migrate": "node -r dotenv/config db/migrate.js && db/setup-test-db.js",
    "db:setup-test": "db/setup-test-db.js",
    "db:destroy": "docker rm -fv spark-db",
    "db:reset": "pnpm run db:destroy && pnpm run db:start",
  }
}
```

In this setup, I used two databases - one for development and debugging, the other for testing. That way, `npm test` did not interfere with the data seen and written by `npm start`.

Reply:
> I can see that on another page (https://postgresapp.com/) they put "your system user name" instead of "your user name". Do you think that's better?

Yes, I think "your system user name" is more clear.

> For that you can set the environment variable DATABASE_URL, as is done in deployments. Do you think that's good enough for your dev flow? If so, shall we document it?

That should work, although I will probably need to figure out how to set that env var for spark-api project only.

Yes please, let's mention this env var in the documentation for developers.

How do you feel about using `node -r dotenv/config` so that I can put `DATABASE_URL` into `.env` file?

> I believe db user management is out of scope for this project. The user needs to be able to do anything it wants with the given database. It doesn't need to create users or perform other tasks. Shall we document that?

Sure, I agree DB management is out of scope of this project. It was not clear to me from the docs what I need to do to setup the DB for spark-api.

I think the docs should say that the user and the database must already exist in the DB server and the user must have full access to this particular database, especially permissions to create and alter tables. WDYT?

> > It would be great to have a script I can run to set everything up for me.
>
> We can add that if you want 👍 As long as we don't make it a requirement, I don't want to have to run docker.

We can support both ways:
- A script that runs as `root` or `admin` and sets up everything needed on the local PG server.
- Another script to setup the DB via Docker - this should be pretty much a one-liner.

Let's not worry about the Docker workflow for now. I'll figure it out when I need to run this project locally.

Reply:
> How do you feel about using node -r dotenv/config so that I can put DATABASE_URL into .env file?

Sgtm! However I would prefer to use dotenv as a lib

Cool, addressed all remaining points in https://github.com/filecoin-station/spark-api/pull/10